src/vanilla/Model/CompositeTypeCs.cs

```diff
@@ -159,13 +159,20 @@ public string Signature
                     foreach (PropertyCs property in Parameters.Where(p => !p.UnderlyingProperty.IsConstant).Select(p => p.UnderlyingProperty))
                     {
                         string format = (property.IsRequired ? "{0} {1}" : "{0} {1} = default({0})");
+                        
+                        // we need some special signatures for extensible enums
+                        var modelTypeName = 
+                            (property.ModelType is EnumType && (property.ModelType as EnumType).ModelAsExtensible)?
+                                 (property.ModelType as EnumType).ClassName : property.ModelTypeName;

```

Comment:
just to raise the point again: why don't we make `ModelTypeName` return `ClassName` of the enum? Did you investigate that path? Like this, I'm just super afraid about all other locations using `property.ModelTypeName`! Why don't they need the enums class name?